### ISSUE TITLE
Configurable asset type

### ISSUE THREAD

**Issue by JurreBrandsenInfoSupport** [vuejs/vitepress#3818]
### Is your feature request related to a problem? Please describe.

The problem I encounter is the fact that certain assets (`.pptx`, `.xlsm`, `.xlsx`) do not behave as I would expect. Currently, if you put these files in the` ./docs/public` folder, you get a link to the location. Clicking this link directs you to a 404 page.

```markdown
## Resources
-   [Agile Inception Deck](/documents/agile-inception-deck.pptx)
```

![image](https://github.com/vuejs/vitepress/assets/149962077/71f2f469-07cd-483f-942e-6ea6fcd62f3e)

What I would expect is that pressing the link would open a download screen, as these files cannot be opened in the browser.

---

The file `agile-inception-deck.pptx` is stored in the `./docs/public/documents/` folder.

### Describe the solution you'd like

I'd like to be able to configure assets in the `config.ts` file, where I would be able to configure how these assets should be handled.

Example
```markdown
assetAction {
    pattern: ['**/*.{pptx, xlsm}']
    action: 'download'
}
```

### Describe alternatives you've considered

Adding custom mimetypes to the vite source code as per the instructions of described in [constants.ts](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/constants.ts)

### Additional context

The current workaround for this issue is using the `<a>` tag with `target="_blank"` in the markdown files. i.e., 
```<a href="/documents/agile-inception-deck.pptx" target="_blank">Agile Inception Deck</a>```
instead of  
```[Agile Inception Deck](/documents/agile-inception-deck.pptx)```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

**Comment by brc-dd** [comment]
Try adding this to top of your `.vitepress/config`:

```ts
process.env.VITE_EXTRA_EXTENSIONS = 'pptx,xlsm'
```

**Comment by JurreBrandsenInfoSupport** [comment]
Thank you for the quick response. I've tested both suggestions, and they both seem to be effective. However, I've noticed that when executing npm run docs:build, all xlsx and xlsm files are showing up as dead links. Is this the intended behavior? I'm hesitant to enable the ignoredeadlinks flag in the settings, as it might lead to overlooking genuine dead links elsewhere in the project.

![image](https://github.com/vuejs/vitepress/assets/149962077/a8c0dab4-ed99-451b-a69a-b161bdf309d9)
